test/linalg/diagonal.jl

```diff
@@ -2,12 +2,45 @@
 
 using Base.Test
 import Base.LinAlg: BlasFloat, BlasComplex, SingularException
+import Base: size, linearindexing, LinearSlow, getindex, length, setindex!, ndims
 
 debug = false
 
 n=12 #Size of matrix problem to test
 srand(1)
 
+#we'll use this to test A_ldiv_B!
+type SparseVec{T} <: AbstractVector{T}
```

Comment:
I saw your discussion, but only this morning. It just seems a bit overkill to define a whole new type for this if an existing type actually covers this. Isn't it the method below you are interested in testing?
```jl
julia> @which A_ldiv_B!(Diagonal([1,2,3.0]), sub(randn(3), [1,2,3]))                       
A_ldiv_B!{T}(D::Base.LinAlg.Diagonal{T}, v::AbstractArray{T,1}) at linalg/diagonal.jl:99
```